sled-agent/src/services.rs

```diff
@@ -344,13 +376,43 @@ impl ServiceManager {
                 ServiceType::Nexus { internal_address, external_address } => {
                     info!(self.log, "Setting up Nexus service");
 
+                    // The address of Nexus' external interface is a special
+                    // case; it may be an IPv4 address.
+                    let addr_request =
+                        AddressRequest::new_static(external_address.ip(), None);
+                    running_zone
+                        .ensure_external_address_with_name(
+                            addr_request,
+                            "public",
+                        )
+                        .await?;
+
+                    match external_address.ip() {
```

Comment:
Style: `if let IpAddr::V4(_public_addr4) = external_address.ip() {` might be a bit clearer.

Reply:
I think it's plausible we'll someday want to allow serving Nexus externally over client-allocated IPv6 addresses, but I'll use this shorter syntax in the meantime.